test/pullrequest/e2e_test.go

```diff
@@ -0,0 +1,170 @@
+package pullrequest
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/jenkins-x/go-scm/scm"
+)
+
+var (
+	proxy = flag.Bool("use-fake-scm-proxy", true, "whether to use fake in-memory proxy instead of using real SCM servers")
```

Comment:
im not a huge fan of this being global - i cant figure out a better option but i feel like putting it in an `init` function is slightly better? it's effectively the same :( i dont have anything super useful to say here; i feel like we should be defining all the test flags in one place but then that makes it harder to track down what is doing :woman_shrugging: 

We do have init_test.go which is which is where the other flags are handled.

Reply:
> but we should aim to consolidate these flags into a single "main.go" - in this case e2e_test.go to keep with test file semantics and to keep the flag definitions as close to their usage as possible.

I think this would make sense to me for separate binaries, but afaik every end to end test will pull in this flag as a result of it being declared globally.

The only other option I can think of is to add a build flag around the whole test, in which case it really would be a separate binary, but I think that makes things more confusing in the long run.

I dunno, I don't feel too strongly.

Reply:
Talked about this a bit offline -

For now this is fine because it's the only test in the package. We should be mindful of this flag spreading to other binaries / libraries in the future though.